src/View/ViewableData.php

```diff
@@ -601,7 +612,10 @@ public function cachedCall($fieldName, $arguments = [], $identifier = null)
     public function hasValue($field, $arguments = [], $cache = true)
     {
         $result = $this->obj($field, $arguments, $cache);
-        return $result->exists();
+        if ($result instanceof ViewableData) {
+            return $result->exists();
+        }
+        return (bool) $result;
```

Comment:
This is still needed because as mentioned above the `obj()` method can return arbitrary objects with its current implementation. It's not related to the issue at hand but worth keeping IMO.